### PR TITLE
Allow decorrelation using energy differences rather than dhdl when dhdl is not available

### PR DESCRIPTION
**Pull request by davidlmobley**
Updating, as per issue #29 , to allow decorrelation of data using energy differences rather than dhdl when dhdl is not available.

### CODE REVIEW

**Comment by davidlmobley** [comment]
@limn1 - can you check and see if this resolves your issue before I merge? This code is from @FEPanalysis and I haven't tested it out yet. (Feel free also to look over the differences - click the "3b87e2d" link above - and see if it looks OK to you, if you like).

Thanks and sorry for the delay. I'd been sitting on this.

File: alchemical_analysis/alchemical_analysis.py [hunk 1]
```diff
@@ -170,28 +171,63 @@ def uncorrelate(sta, fin, do_dhdl=False):
...
+            for n in range(n_components):
+               dhdl[k,n,0:N] = dhdlt[k,n,indices]
+
+   if UNCORR_OBSERVABLE == 'dE':
+
+      for k in range(K):
+         ## Sum up over the energy components; notice, that only the relevant data is being used in the third dimension.
+         #dhdl_sum = numpy.sum(dhdlt[k,:,sta[k]:fin[k]], axis=0)
+         # Determine indices of uncorrelated samples from potential autocorrelation analysis at state k
+         # (alternatively, could use the energy differences -- here, we will use total dhdl).
```
Comment (mrshirts) [inline alchemical_analysis/alchemical_analysis.py]: Thanks for taking care of this!  You probably want to clean up this comment so it's clearer -- right now it's a repeat from above.

**Comment by FEPanalysis** [comment]
Did you guys have a chance to test it?

**Comment by davidlmobley** [comment]
@limn1, can you check this on your data so we can merge?

File: alchemical_analysis/alchemical_analysis.py (hunk 1 above)
Comment (davidlmobley) [inline alchemical_analysis/alchemical_analysis.py]: Fixed, thanks.

**Comment by davidlmobley** [comment]
@limn1 - did #41 include this or does this still need to be merged as well? Thanks.

**Comment by davidlmobley** [comment]
Wait, nevermind, now I see - your PR was into this branch. So now that I've merged your branch into here this is ready to merge with main. I'm merging.

**Pull request merged**